irr/include/vector2d.h

```diff
@@ -34,6 +35,9 @@ class vector2d
 	constexpr vector2d(const dimension2d<T> &other) :
 			X(other.Width), Y(other.Height) {}
 
+	constexpr vector2d(const std::array<T, 2> &arr) :
```

Comment:
Maybe mark it explicit.

Reply:
Personally I consider this one of a few places where an implicit conversion wouldn't surprise me, but sure